zilencer/views.py

```diff
@@ -215,22 +234,23 @@ def remote_server_notify_push(
 ) -> HttpResponse:
     server = validate_entity(entity)
 
-    user_id = payload["user_id"]
+    user_identity = UserPushIndentityCompat(payload.get("user_id"), payload.get("user_uuid"))
```

Comment:
This can be for a follow-up PR (certainly a follow-up commit!), but I think it'd be a good for the server to, when receiving such a request, rewrite matching PushDeviceToken entries with just a `user_id` to instead have just the `user_uuid`.  

That way, once a remote Zulip server has upgraded to sending the correct format, it is immediately no longer vulnerable to this issue.